src/Symfony/Component/Cache/Adapter/ArrayAdapter.php

```diff
@@ -55,12 +55,22 @@ function ($key, $value, $isHit) use ($defaultLifetime) {
      */
     public function getItem($key)
     {
-        if (!$isHit = $this->hasItem($key)) {
+        $isHit = $this->hasItem($key);
+        try {
+            if (!$isHit) {
+                $value = null;
+            } elseif (!$this->storeSerialized) {
+                $value = $this->values[$key];
+            } elseif ('b:0;' === $value = $this->values[$key]) {
+                $value = false;
+            } elseif (false === $value = unserialize($value)) {
+                $value = null;
+                $isHit = false;
+            }
+        } catch (\Exception $e) {
+            CacheItem::log($this->logger, 'Failed to unserialize key "{key}"', array('key' => $key, 'exception' => $e));
```

Comment:
There is no parent here, and calling the same implementation would call ini_set twice for something that can't happen because as you said, this is not cross reqs